lib/configure/criteria.js

```diff
@@ -1,22 +1,96 @@
 var _ = require('lodash')
+var glob = require('glob')
 
-module.exports = function (locator) {
-  var parts
-  if (_.includes(locator, '#')) {
-    parts = locator.split('#')
-    return {
-      glob: parts[0],
-      name: parts[1]
-    }
-  } else if (_.includes(locator, ':')) {
-    parts = locator.split(':')
-    return {
-      glob: parts[0],
-      lineNumber: parts[1]
-    }
-  } else {
-    return {
-      glob: locator
-    }
+module.exports = function (testLocators, userOptions) {
+  userOptions = userOptions || {}
+  testLocators = _.castArray(testLocators)
+
+  // Break the test locator glob patterns apart from filter suffixes
+  var testCriteria = _.map(testLocators, function (testLocator) {
+    return globCriterion(testLocator)
+  })
+
+  // If we have global test name filters, add it to the criteria as though
+  // it were provided with each glob. This allows us to eliminate duplicate
+  // name filters in the collected criteria
+  testCriteria = _.map(testCriteria, function (globCriterion) {
+    return applyNameFilters(globCriterion, userOptions.example)
+  })
+
+  // Expand globbed critera into criteria with file paths. Each resulting
+  // criterion receives a copy of the criteria associated to the glob.
+  testCriteria = _.flatMap(testCriteria, function (globCriterion) {
+    return pathCriteria(globCriterion)
+  })
+
+  // Consolidate criteria with duplicate file paths into a single criterion
+  // file paths with arrays of collected  line / name filter values
+  testCriteria = consolidatedPathCriteria(testCriteria)
```

Comment:
Would you be open to refactoring this so we're not re-assigning the same var over and over in the same function? Seems like the sort of thing that'd `best(be(several(calls()))))` to named functions, hopefully obviating the comments

Reply:
@searls I've refactored https://github.com/testdouble/teenytest/blob/92ce2b86d66845e4c0dbd5df59127b548cff2809/lib/configure/criteria.js. If anything still seems like it should be changed (or just rubs you the wrong way) let me know.